rust/src/ops/max/mod.rs

```diff
@@ -0,0 +1,111 @@
+use std::cmp::{Ord, PartialEq};
+use std::marker::PhantomData;
+
+use num_traits::cast::{ToPrimitive, NumCast};
+
+use alga::general::AbstractMagma;
+use alga::general::AbstractMonoid;
+use alga::general::AbstractSemigroup;
+use alga::general::Identity;
+use alga::general::Operator;
+
+use super::AggregateOperator;
+use super::AggregateMonoid;
+
+// We need a generic way to talk about a type's minimum 
+// value. Unfortunately, we'll need a new implementation 
+// for every type we want to support. I don't know of a 
+// way around this yet.
+pub trait Min {
```

Comment:
It might be possible to later use [Bounded](https://docs.rs/num-traits/0.2.14/num_traits/bounds/trait.Bounded.html) of `num-traits`. It is min()/max() implemented for a lot of scalars. Otherwise maybe it is possible to write a macro for deriving it.

Reply:
I'm going to try that now. That looks like exactly the kind of thing I couldn't find before.

Reply:
Works great, applied.